### DISCUSSION TITLE
Prevent parent loader call when child route query params change

### DISCUSSION THREAD

**Discussion by iamzee** [remix-run/remix#7609] [Q&A] [answered]
Lets say i have two routes

```
routes.parent.tsx
routes.parent.child.tsx
```

I have some paginated table in `routes.parent.child.tsx` which i handle using query params `?offset=0`
But when queryparams change the parent loader is also called along with child loader.

How to prevent parent loader call in this case..

I have tried `shouldRevalidate` on parent...but it is not being called on query param change

child route is something like this
```javascript
// routes.parent.child.tsx

export const laoder = () => something;
export default function ChlldRoute() {
    return (
    <Link to="?offset=20">Next</Link>
  );
}

```

**Comment by kiliman** [comment, marked as answer]
I created an example on StackBlitz that appears to work.

You can verify by opening DevConsole and viewing console logs.

⚡️ https://stackblitz.com/edit/remix-run-remix-yy9xue?file=app%2Froutes%2Fparent.tsx,app%2Froutes%2Fparent.child.tsx

**Reply by iamzee** [reply]
Got it, thanks @kiliman

**Reply by kiliman** [reply]
Did you figure out your issue?

**Reply by iamzee** [reply]
Yes, @kiliman 
I was doing `console.log` inside `shouldRevalidate` and was checking the terminal console instead of browser console. 😞 
I thought `shouldRevalidate` is called on the server...

**Reply by kiliman** [reply]
Yeah, I get confused, too. The whole point of `shouldRevalidate` is to *prevent* a call from the client to the server. 

Here's a handy table
|export          |server|client|
|----------------|------|------|
|action          |  ✅  |  ❌   |
|loader          |  ✅  |  ❌   |
|headers         |  ✅  |  ❌   |
|handle          |  ✅  |  ✅   |
|links           |  ✅  |  ✅   |
|meta            |  ✅  |  ✅   |
|shouldRevalidate|  ❌  |  ✅   |

**Reply by iamzee** [reply]
Thanks @kiliman 🙌 This is useful.
